OpenTrons/opentrons-api
# feat(app, app-shell, app-shell-odd): add a new config for initial setup workflow

# Overview
add targetPath, a new config value for the initial setup flow

Currently the ODD app always shows RobotDashboard even thought the initial setup isn't done yet. This PR is adding `targetPath` to config to solve that issue.

the default value is `/welcome` screen -> the app shows -> welcome screen
Then a user needs to finish the initial setup flow (network setup -> software update -> name robot)
After the software update, the robot needs to reboot and the app needs to show Name screen for this the app update `targetPath` to `/robot-settings/rename-robot` from `/welcome`.
Finally after naming a robot, the app change `targetPath` to `/dashboard`.

Unfortunately, right now we don't have a way to check these process with UI since the software update is technically downgrade for this branch.

What we can do is to check `Config.json` value. The details are in `Test Plan` 

close RCORE-386


# Test Plan
1. Push this branch to a robot / dev kit
2. Access to the robot/dev kit via screen or ssh
3. Go to ODD folder with `cd /data/ODD` and make sure that there is `Config.json` file.
4. Check the value of `Config.json`
In `Config.json`, there is OnDeviceDisplaySettings: targetPath and its value should be `welcome`
5. The app shows Welcome screen instead of RobotDashboard
6. Do the initial setup (network settings and update softeare)
7. Check `Config.json` and the value of `targetPath` is updated to `/robot-settings/rename-robot`
8. Tap ODD menu -> `Rename Robot` and name a robot
9. Check `Config.json` and the value of `targetPath` is updated to `/dashboard`

For this change, we will need to add additional parameter to use on-device mode since there is no way to update the `targetPath` from on-device mode UI.
```zsh
make -C app dev OT_APP_IS_ON_DEVICE=1 OT_APP_ON_DEVICE_DISPLAY_SETTINGS__TARGET_PATH=/dashboard
```
Or just edit the `config.json` directly.
path is `/Users/${yourUserName}/Library/Application Support/Opentrons`



# Changelog
- add a new config value to `config.OnDeviceDisplaySettings`
- update OnDeviceDisplayApp
- update tests


# Review requests



# Risk assessment
low
